code/go/0chain.net/chaincore/chain/state/state_context.go

```diff
@@ -58,6 +58,8 @@ type StateContextI interface {
 	EmitError(error)
 	GetEvents() []event.Event   // cannot use in smart contracts or REST endpoints
 	GetEventDB() *event.EventDb // do not use in smart contracts can use in REST endpoints
+	GetCurrentRewardRound(period int64) int64
+	GetPreviousRewardRound(period int64) int64
```

Comment:
I cannot see why these two need to be in StateContextI. Cannot you just have a local function in stroagesc?
```go
func  GetCurrentRewardRound(period int64, balances cstate.StateContextI) int64 {
	extra := balances.GetBlock().Round % period
	return sc.block.Round - extra
}
```
Adding to StateContextI creates a maintenance overhead as it is used in so many places.